v3/test/configuration_test_framework_test.go

```diff
@@ -0,0 +1,203 @@
+package test
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"sync"
+	"testing"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/util"
+
+	"github.com/zmap/zlint/v3/lint"
+)
+
+type caCommonNameMissing struct {
+	BeerHall string
+	Working  *lint.CABFBaselineRequirementsConfig
+}
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:          "e_ca_common_name_missing2",
+		Description:   "CA Certificates common name MUST be included.",
+		Citation:      "BRs: 7.1.4.3.1",
+		Source:        lint.CABFBaselineRequirements,
+		EffectiveDate: util.CABV148Date,
+		Lint:          NewCaCommonNameMissing,
+	})
+}
+
+func (l *caCommonNameMissing) Configure() interface{} {
+	return l
+}
+
+func NewCaCommonNameMissing() lint.LintInterface {
+	return &caCommonNameMissing{}
+}
+
+func (l *caCommonNameMissing) CheckApplies(c *x509.Certificate) bool {
+	return util.IsCACert(c)
+}
+
+func (l *caCommonNameMissing) Execute(c *x509.Certificate) *lint.LintResult {
+	if c.Subject.CommonName == "" {
+		return &lint.LintResult{Status: lint.Error, Details: l.BeerHall}
+	} else {
+		return &lint.LintResult{Status: lint.Pass, Details: l.BeerHall}
+	}
+}
+
+func TestCaCommonNameMissing(t *testing.T) {
+	inputPath := "caCommonNameMissing.pem"
+	expected := lint.Error
+	out := TestLint("e_ca_common_name_missing2", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
+
+func TestCaCommonNameNotMissing(t *testing.T) {
+	inputPath := "caCommonNameNotMissing.pem"
+	expected := lint.Pass
+	out := TestLint("e_ca_common_name_missing2", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
+
+func TestCaCommonNameNotMissing2(t *testing.T) {
+	inputPath := "caCommonNameNotMissing.pem"
+	expected := lint.Pass
+	config := `
+[e_ca_common_name_missing2]
+BeerHall = "liedershousen"
+`
+	out := TestLintWithConfig("e_ca_common_name_missing2", inputPath, config)
+	if out.Details != "liedershousen" {
+		panic("noooooooooooo")
```

Comment:
Why panic instead of `t.Errorf(...)` here? (And a few other places below.)

Reply:
Simply some first-pass yolo test code that I missed when sprucing everything up :smiley: